### PR TITLE
musly: init at unstable-2017-04-26

### PR DESCRIPTION
**Pull request by ggPeti**
###### Motivation for this change
Musly is a fast and effective open source audio similarity tool.
This package is also the first time it's ported to macOS.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1765,6 +1765,11 @@
     github = "Gerschtli";
     name = "Tobias Happ";
   };
+  ggpeti = {
```
Comment (etu) [inline maintainers/maintainer-list.nix]: This should be part of a separate commit. Like `maintainers/maintainer-list: Add ggpeti as maintainer`

File: pkgs/applications/audio/musly/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, cmake, eigen, libav_all }:
+stdenv.mkDerivation rec {
+  name = "musly-${version}";
```
Comment (etu) [inline pkgs/applications/audio/musly/default.nix]: `pname = "musly"`;

File: pkgs/applications/audio/musly/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  version = "0.2";
```
Comment (etu) [inline pkgs/applications/audio/musly/default.nix]: There's no tag named 0.2, since you pick a random git revision you should put the commit-date here instead, the form `unstable-YYYY-MM-DD`  is usually quite accepted.

File: pkgs/applications/audio/musly/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner = "dominikschnitzer";
+    repo = "musly";
```
Comment (etu) [inline pkgs/applications/audio/musly/default.nix]: `repo = pname;`

File: pkgs/applications/audio/musly/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    cmake -DCMAKE_INSTALL_PREFIX=$out $src
+  '';
+  fixupPhase = if stdenv.isDarwin then ''
+    install_name_tool -change libmusly.dylib $out/lib/libmusly.dylib $out/bin/musly
+    install_name_tool -change libmusly_resample.dylib $out/lib/libmusly_resample.dylib $out/bin/musly
+    install_name_tool -change libmusly_resample.dylib $out/lib/libmusly_resample.dylib $out/lib/libmusly.dylib
+  '' else "";
+
+  meta = with stdenv.lib; {
+    homepage = http://www.musly.org;
```
Comment (etu) [inline pkgs/applications/audio/musly/default.nix]: Their website supports https, so that's preferred by most people on the internet nowadays.

File: pkgs/applications/audio/musly/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      Musly analyzes the the audio signal of music pieces to estimate their similarity.
+      No meta-data about the music piece is included in the similarity estimation.
+      To use Musly in your application, have a look at the library documentation
+      or try the command line application included in the package and start generating
+      some automatic music playlists right away.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ ggpeti ];
+  };
+}
```
Comment (etu) [inline pkgs/applications/audio/musly/default.nix]: And while you're at it editing the file you may add a single newline character at the end of file to get rid of this red arrow.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (ggPeti) [reply maintainers/maintainer-list.nix]: I wouldn't want to force push to the PR as it broke checks the other day. So I'll have to leave it there, thanks for the tip though.

File: pkgs/applications/audio/musly/default.nix (hunk 4 above)
Comment (ggPeti) [reply pkgs/applications/audio/musly/default.nix]: I prefer leaving that one unabstracted so that the `fetchFromGitHub` block is copiable on its own.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (etu) [reply maintainers/maintainer-list.nix]: @ggPeti If it breaks checks we'd just have to re-eval them, and breaking them is probably not your fault. We'd prefer if it's separate since the commits become more independent.

**Comment by etu** [comment]
@ggPeti We'd also prefer if you squash your commits with the license fix.

@GrahamcOfBorg build musly

File: pkgs/applications/audio/musly/default.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+stdenv.mkDerivation rec {
+  pname = "musly";
+  version = "unstable-2017-04-26";
+  src = fetchFromGitHub {
+    owner = "dominikschnitzer";
+    repo = "musly";
+    rev = "f911eacbbe0b39ebe87cb37d0caef09632fa40d6";
+    sha256 = "1q42wvdwy2pac7bhfraqqj2czw7w2m33ms3ifjl8phm7d87i8825";
+  };
+  buildInputs = [ cmake eigen (libav_all.override { vaapiSupport = stdenv.isLinux; }).libav_11 ];
```
Comment (jtojnar) [inline pkgs/applications/audio/musly/default.nix]: `cmake` should go to `nativeBuildInputs` as it is run during the build.

File: pkgs/applications/audio/musly/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  preBuild = ''
```
Comment (jtojnar) [inline pkgs/applications/audio/musly/default.nix]: This whole attribute is not necessary at all – adding `cmake` to build inputs will do that automatically.

File: pkgs/applications/audio/musly/default.nix [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    longDescription = ''
+      Musly analyzes the the audio signal of music pieces to estimate their similarity.
+      No meta-data about the music piece is included in the similarity estimation.
+      To use Musly in your application, have a look at the library documentation
+      or try the command line application included in the package and start generating
+      some automatic music playlists right away.
+    '';
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ ggpeti ];
+  };
```
Comment (jtojnar) [inline pkgs/applications/audio/musly/default.nix]: Please add `platforms`.

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by ggPeti** [comment]
@jtojnar @etu pls approve

**Comment by etu** [comment]
@GrahamcOfBorg build musly

**Comment by ggPeti** [comment]
I'm not sure why the checks say "No attempt" - am I still doing something wrong?

File: pkgs/applications/audio/musly/default.nix [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    longDescription = ''
+      Musly analyzes the the audio signal of music pieces to estimate their similarity.
+      No meta-data about the music piece is included in the similarity estimation.
+      To use Musly in your application, have a look at the library documentation
+      or try the command line application included in the package and start generating
+      some automatic music playlists right away.
+    '';
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ ggpeti ];
+    platforms = with platforms; [ darwin linux ];
```
Comment (jtojnar) [inline pkgs/applications/audio/musly/default.nix]: Try

```suggestion
    platforms = with platforms; darwin ++ linux;
```

**Comment by etu** [comment]
@GrahamcOfBorg build musly

**Comment by ggPeti** [comment]
Everything seems to be fine

**Comment by brainrape** [comment]
Built it, ran it. Seems fine.

**Comment by etu** [comment]
@ggPeti Thank you very much for your contribution and patient to get everything together :)
